spec/lib/fastly_nsq/rake_task_spec.rb

```diff
@@ -3,71 +3,124 @@
 
 RSpec.describe MessageQueue::RakeTask do
   before(:each) do
-    Rake::Task['begin_listening'].clear if Rake::Task.task_defined?('begin_listening')
+    Rake::Task.clear
+    allow_any_instance_of(MessageQueue::RakeTask).to receive(:output) { nil }
   end
 
   describe 'when defining tasks' do
-    it 'creates a begin_listening task' do
-      MessageQueue::RakeTask.new
+    context 'when no task name is provided' do
+      it 'creates a task with the default name' do
+        default_task_name = 'begin_listening'
 
-      allow_any_instance_of(MessageQueue::RakeTask).to receive(:output) { nil }
+        MessageQueue::RakeTask.new
+        defined_tasks = Rake::Task.tasks
+        first_task_name = defined_tasks.first.name
 
-      is_defined = Rake::Task.task_defined?(:begin_listening)
-      expect(is_defined).to be_truthy
+        expect(first_task_name).to eq default_task_name
+      end
     end
 
-    it 'creates a named task' do
-      MessageQueue::RakeTask.new(:test_name)
+    context 'when a task name is passed in' do
+      it 'creates a task with the provided name' do
+        task_name = 'test_name'
 
-      allow_any_instance_of(MessageQueue::RakeTask).to receive(:output) { nil }
+        MessageQueue::RakeTask.new(task_name.to_sym)
+        defined_tasks = Rake::Task.tasks
+        first_task_name = defined_tasks.first.name
 
-      is_defined = Rake::Task.task_defined?(:test_name)
-      expect(is_defined).to be_truthy
+        expect(first_task_name).to eq task_name
+      end
     end
   end
 
   describe 'when running tasks' do
-    it 'runs with inline options defined' do
-      options = { topic: 'dwarf', channel: 'star' }
-      task = MessageQueue::RakeTask.new(:begin_listening, [:topic, :channel])
+    context 'when multiple topics are defined' do
+      it 'listens to both channels' do
```

Comment:
I think you mixed up `channel` and `topic` a bit in the wording here.
Additionally, it doesn't look to me like this goes out of its whay to check that the channel is correct or even being checked...
Perhaps:
```ruby
context 'when multiple topics are defined' do
  it 'listens to both on the same chanel' do
```
or: